docs/csharp/language-reference/keywords/try-finally.md

```diff
@@ -14,7 +14,7 @@ ms.assetid: c27623fb-7261-4464-862c-7a369d3c8f0a
 
 By using a `finally` block, you can clean up any resources that are allocated in a [try](try-catch.md) block, and you can run code even if an exception occurs in the `try` block. Typically, the statements of a `finally` block run when control leaves a `try` statement. The transfer of control can occur as a result of normal execution, of execution of a `break`, `continue`, `goto`, or `return` statement, or of propagation of an exception out of the `try` statement.
 
-Within a handled exception, the associated `finally` block is guaranteed to be run. However, if the exception is unhandled, execution of the `finally` block is dependent on how the exception unwind operation is triggered. That, in turn, is dependent on how your computer is set up.
+Within a handled exception, the associated `finally` block is guaranteed to be run. However, if the exception is unhandled, execution of the `finally` block is dependent on how the exception unwind operation is triggered. That, in turn, is dependent on how your computer is set up. The only cases where `finally` clauses don't run involve a program being immediately stopped. An example of this would be when `InvalidProgramException` gets thrown because of the IL statements being corrupt. On most operating systems, reasonable resource cleanup will take place as part of stopping and unloading the process.
```

Comment:
If a `try/finally` statement is written without `catch`. Does finally block run if the try throws an exception?

If not, this should also be added as a case.

Reply:
In a `try`/`finally`, yes, the finally runs if the `try` throws an exception